### PR TITLE
virtio_fs: new case of folder testing for nfs backend

### PR DESCRIPTION
**Pull request by xiagao**
1. add rename and delete folder steps.
2. support nfs backend test scenario.
ID: 1993725
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
(1/5) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.q35: PASS (92.37 s)
 (2/5) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_nfs.with_cache.auto.q35: ERROR: Shell command failed: 'rm -rf /mnt/myfs/test_folder_tmp'    (status: 1,    output: "rm: cannot remove '/mnt/myfs/test_folder_tmp': Directory not empty\n") (93.02 s)  ---need to debug if product issue.
 (3/5) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_multi_fs_sources.with_cache.auto.q35: PASS (235.46 s)
 (4/5) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.q35: PASS (177.05 s)
 (5/5) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_nfs.with_cache.auto.q35: FAIL: The folder are not deleted. (101.70 s)  --known bz

**Comment by xiagao** [comment]
@YongxueHong @zhenyzha Could you help review? Thanks.

File: qemu/tests/cfg/virtio_fs_share_data.cfg [hunk 1]
```diff
@@ -103,22 +103,37 @@
...
                 cmd_copy_file = "copy %s ${test_folder}"
-                cmd_del_folder = "rd /q /s ${test_folder}"
+                cmd_rename_folder = "%s && rename test_folder test_folder_tmp"
+                cmd_del_folder = "rd /q /s ${test_folder}_tmp"
+                cmd_check_folder = "dir ${test_folder}_tmp"
             default..with_cache.none:
                 io_timeout = 600
             variants:
                 - @default:
+                - with_nfs:
```
Comment (YongxueHong) [inline qemu/tests/cfg/virtio_fs_share_data.cfg:127]: Is there a difference between `- with_nfs_source` and `- with_nfs`?
Can we merge them into one case?

Comment (xiagao) [reply qemu/tests/cfg/virtio_fs_share_data.cfg:127]: Actually they are different test scenarios,the previous one is for basic file/folder test. The other one is for xfstest.

Comment (YongxueHong) [reply qemu/tests/cfg/virtio_fs_share_data.cfg:127]: Okay, but I think we still can do it, like:
```
    variants: # case about testing source type
        - with_local_source:
            ...
        - with_nfs_source:
            ...
    variants: # case about testing virtiofsd option
        - with_cache:
            ...
        - with_no_writeback:
            ...
    variants: # case about testing tool
        - @default:
            ...
            variants:
                - @default:
                - with_multi_fs_sources:
                    ...
        - run_stress:
            variants:
                - with_fio:
                    ...
                - with_pjdfstest:
                    ...
                - with_xfstest:
                    ...
```
This is maybe more readable.

Comment (zhenyzha) [reply qemu/tests/cfg/virtio_fs_share_data.cfg:127]: In fact, I also feel that they can be merged, one is local testing, the other is nfs backend test.
Similar to yhong's opinion, please feel free to share your opinion. 

`        - with_local_source:
            ...
        - with_nfs_source:
            variants:
                - @default:
                - with_xfstest
`

**Comment by xiagao** [comment]
Of course it's better to rearrange the variants, but it will be a big change,including cfg file structure change, parameters are redefined such as filesystems defined in different scenarios and py file need changed accordingly.
The impact I can guess is all virtiofs test case name will be changed.
@YongxueHong @zhenyzha 
If you don't mind of the change, I will close this pr and rewrite a new one to support the overall change。

**Pull request closed without merging**